src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Parameters/SearchParameterValidator.cs

```diff
@@ -75,18 +74,44 @@ public async Task ValidateSearchParamterInput(SearchParameter searchParam, strin
             {
                 try
                 {
-                    _searchParameterDefinitionManager.GetSearchParameter(new Uri(searchParam.Url));
-
-                    // If a post, then it is a creation of a new search parameter
-                    // only allow this if no other parameters exist with the same Uri
-                    if (method.Equals(HttpPostName, StringComparison.OrdinalIgnoreCase))
+                    // If a search parameter with the same uri exists already
+                    if (_searchParameterDefinitionManager.TryGetSearchParameter(new Uri(searchParam.Url), out _))
+                    {
+                        // And if this is a request to create a new search parameter
+                        if (method.Equals(HttpPostName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            // We have a conflict
+                            validationFailures.Add(
+                                new ValidationFailure(
+                                    nameof(searchParam.Url),
+                                    string.Format(Resources.SearchParameterDefinitionDuplicatedEntry, searchParam.Url)));
+                        }
+                    }
+                    else
                     {
-                        // if no exception is thrown, then the search parameter with the same Uri was found
-                        // and this is a conflict
-                        validationFailures.Add(
-                            new ValidationFailure(
-                                nameof(searchParam.Url),
-                                string.Format(Resources.SearchParameterDefinitionDuplicatedEntry, searchParam.Url)));
+                        // Otherwise, no search parameters with a matching uri exist
+                        // Ensure this isn't a request to modify an existing parameter
+                        if (method.Equals(HttpPutName, StringComparison.OrdinalIgnoreCase) ||
+                            method.Equals(HttpDeleteName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            validationFailures.Add(
+                                new ValidationFailure(
+                                    nameof(searchParam.Url),
+                                    string.Format(Resources.SearchParameterDefinitionNotFound, searchParam.Url)));
+                        }
+
+                        // Ensure the new search parameter's code value does not already exist for its base type(s)
```

Comment:
Is it possible for a PUT request to update an existing valid search parameter with wrong code information? If so, shouldn't we do this same validation for the following scenario - SearchParameter with same uri already exists and it is a PUT request (the if condition on line 78)?

Reply:
Good catch, thanks :). Fixed and added a new test case in 5258560.